src/sections/home/Home.Events.tsx

```diff
@@ -1,57 +1,43 @@
-import Image from "next/image"
-import { useState, useEffect } from "react"
-import { EventsArray, EventObject, getAllEvents } from "src/api/EventsAPI"
-import { isURL, getAbsoluteURL, getDateTime } from "src/utils"
+import { useState } from "react";
+import { EventObject } from "src/api/EventsAPI";
+import { isURL, getAbsoluteURL, getDateTime } from "src/utils";
 
-const HomeEvents: React.FC = () => {
-  const [events, setEvents] = useState<EventsArray>()
-  const [dragging, toggleDragging] = useState(false)
-
-  const updateEvents = async (): Promise<void> => {
-    const eventsArray: EventsArray | undefined = await getAllEvents()
-    if (eventsArray === undefined) {
-      setEvents(new Array<EventObject>())
-    }
-    setEvents(eventsArray)
-  }
-
-  useEffect(() => {
-    if (!events) {
-      updateEvents()
-    }
-  }, [events])
+const HomeEvents: React.FC<{ events: Array<EventObject> }> = ({ events }) => {
+  const [dragging, toggleDragging] = useState(false);
 
   const handleMove = (e: any) => {
     if (!dragging) {
-      return
+      return;
     }
     const em = parseFloat(
       window.getComputedStyle(
         document.getElementsByClassName("home__events__grid__container")[0]
       ).fontSize
-    )
-    const isTouch = e.movementX === ""
-    let deltaX = 0
+    );
+    const isTouch = e.movementX === "";
+    let deltaX = 0;
     if (isTouch) {
-      deltaX = e.touches[0].clientX
+      deltaX = e.touches[0].clientX;
     } else {
-      deltaX = -1 * e.movementX
+      deltaX = -1 * e.movementX;
     }
     const firstEvent = document.getElementsByClassName(
       "event"
-    )[0] as HTMLElement
-    const cardWidth = em * 20
-    const minMargin = events ? em - events.length * 20 * em : cardWidth
-    const oldmargin = parseFloat(window.getComputedStyle(firstEvent).marginLeft)
-    const newmargin = oldmargin - deltaX
+    )[0] as HTMLElement;
+    const cardWidth = em * 20;
+    const minMargin = events ? em - events.length * 20 * em : cardWidth;
+    const oldmargin = parseFloat(
+      window.getComputedStyle(firstEvent).marginLeft
+    );
+    const newmargin = oldmargin - deltaX;
     if (newmargin >= minMargin && newmargin <= em) {
-      firstEvent.style.marginLeft = `${oldmargin - deltaX}px`
+      firstEvent.style.marginLeft = `${oldmargin - deltaX}px`;
     }
-  }
+  };
```

Comment:
Can you add a comment just explaining the approach for what this code is supposed to do - I don't think it's obvious what it does, and since you're working in this area to clean it up, it'd be helpful to have a comment to help future maintainers know what's going on here

Reply:
LOL i have no clue what that code does tbh. i removed it cause it looked useless but then reverted the commit in case its handling some weird edge case ?? i just had prettier enabled but didnt touch it

Reply:
@steets250 do you know what it does

Reply:
we'll look at this later, but for now @farisashai go ahead and delete this section since it doesn't seem to break anything.